src/subdomains/generic/user/models/auth/auth.controller.ts

```diff
@@ -61,10 +69,25 @@ export class AuthController {
   }
 
   @Get('mail/confirm')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
   @ApiExcludeEndpoint()
-  async executeMerge(@Query('code') code: string, @Res() res: Response): Promise<void> {
+  @ApiOkResponse({ type: MergeRedirectDto })
+  async executeMerge(
+    @GetJwt() jwt: JwtPayload,
+    @Query('code') code: string,
+    @RealIP() ip: string,
+  ): Promise<MergeRedirectDto> {
     const { master } = await this.mergeService.executeMerge(code);
-    res.redirect(master.kycUrl);
+    const newUser = await this.userRepo.findOne({
+      where: { userData: { id: master.id }, address: jwt.address },
+      relations: { userData: true, wallet: true },
+    });
+
+    return {
+      redirectUrl: master.kycUrl,
+      accessToken: this.authService.generateUserToken(newUser, ip),
+    };
   }
```

Comment:
Ok, I see some issues here:
- The user will click a link he received by mail. Therefore he might not be logged in in the Services
   - We cannot have an auth guard
   - We can only return an access token, if the user came with one
   - Maybe we should remove the `redirectUrl` in the response and just return the master's KYC code and the JWT, if the user was logged in
- The User might have an "account-only" account, which means he has no user
   - In this case (there is no `user` in the JWT) use `generateAccountToken`